tools/azsdk-cli/Azure.Sdk.Tools.Cli.Tests/MockServices/MockGitHubService.cs

```diff
@@ -91,6 +96,73 @@ public Task<Issue> GetIssueAsync(string repoOwner, string repoName, int issueNum
             return Task.FromResult<IReadOnlyList<RepositoryContent>?>(contents.AsReadOnly());
         }
 
+        public Task<IReadOnlyList<RepositoryContent>?> GetContentsAsync(string owner, string repoName, string path, string branch)
+        {
+            // Handle specific test scenarios for branch-specific requests
+            if (path == "non-existent-path")
+            {
+                return Task.FromResult<IReadOnlyList<RepositoryContent>?>(null);
+            }
+
+            if (path == "empty-directory")
+            {
+                return Task.FromResult<IReadOnlyList<RepositoryContent>?>(new List<RepositoryContent>().AsReadOnly());
+            }
+
+            // Handle single file requests (when specific files are requested)
+            if (path.Contains("/") && path.EndsWith(".md"))
+            {
+                var fileName = System.IO.Path.GetFileName(path);
+                var content = CreateMockRepositoryContent(fileName, path, "test");
+                return Task.FromResult<IReadOnlyList<RepositoryContent>?>(new List<RepositoryContent> { content }.AsReadOnly());
+            }
+
+            // Default: Return mock directory listing for .github/prompts or similar paths
+            var contents = new List<RepositoryContent>
+            {
+                CreateMockRepositoryContent("README.md", ".github/prompts/README.md", "test"), 
+                CreateMockRepositoryContent("prompt1.md", ".github/prompts/prompt1.md", "test"), 
+                CreateMockRepositoryContent("prompt2.md", ".github/prompts/prompt2.md", "test")  
+            };
+
+            return Task.FromResult<IReadOnlyList<RepositoryContent>?>(contents.AsReadOnly());
+        }
+
+        public Task<string> UpdateFileAsync(string owner, string repoName, string path, string message, string content, string sha, string branch)
+        {
+            // Create a mock RepositoryContentChangeSet
+            var mockContent = CreateMockRepositoryContent(System.IO.Path.GetFileName(path), path, content);
+            var mockCommit = CreateMockCommit(message, sha);
+
+            var changeSet = $"Updated {path} with commit message {mockCommit}";
+
+            return Task.FromResult(changeSet);
```

Comment:
```suggestion
            throw new NotImplementedException();
```

I don't think this is used anywhere.